libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4908,12 +4908,11 @@ void GCS_MAVLINK::convert_COMMAND_LONG_to_COMMAND_INT(const mavlink_command_long
     out.param2 = in.param2;
     out.param3 = in.param3;
     out.param4 = in.param4;
+    out.x = isnan(in.param5) ? 0 : in.param5;
+    out.y = isnan(in.param6) ? 0 : in.param6;
     if (command_long_stores_location((MAV_CMD)in.command)) {
-        out.x = in.param5 *1e7;
-        out.y = in.param6 *1e7;
-    } else {
-        out.x = in.param5;
-        out.y = in.param6;
+        out.x = out.x *1e7;
```

Comment:
that won't work, you'll end up rounded to the nearest degree

Reply:
Fixed.  Or perhaps broken differently.